lib/srv/db/cloud/iam.go

```diff
@@ -298,13 +296,11 @@ func (c *IAM) processTask(ctx context.Context, task iamTask) error {
 	configurator, err := c.getAWSConfigurator(ctx, task.database)
 	if err != nil {
 		c.iamPolicyStatus.Store(task.database.GetName(), types.IAMPolicyStatus_IAM_POLICY_STATUS_FAILED)
-		if errors.Is(trace.Unwrap(err), credentials.ErrNoValidProvidersFoundInChain) {
-			c.logger.WarnContext(ctx, "No AWS credentials provider, Skipping IAM task for database",
-				"database", task.database.GetName(),
-			)
-			return nil
-		}
-		return trace.Wrap(err)
+		c.logger.WarnContext(ctx, "Failed to load AWS IAM configurator, skipping IAM task for database",
+			"database", task.database.GetName(),
+			"error", err,
+		)
+		return nil
```

Comment:
have we tried these scenarios with sdk v2?

Reply:
for example, it may be failing here?
https://github.com/gravitational/teleport/blob/c0b3e62f25301d17859ed05a4d34f5d34af6f75b/lib/srv/db/cloud/iam.go#L245-L253

not worried about the log, but want to make sure we don't swallow the error messages in non-valid-provider cases